nodeadm/cmd/nodeadm/init/environment.go

```diff
@@ -0,0 +1,81 @@
+package init
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"go.uber.org/zap"
+
+	"github.com/awslabs/amazon-eks-ami/nodeadm/internal/api"
+)
+
+const (
+	systemdEnvironmentConfPath = "/etc/systemd/system.conf.d/environment.conf"
+)
+
+// making environment variables available system-wide to all processes and services.
+func handleSystemEnvironmentVariables(log *zap.Logger, instanceOpts api.InstanceOptions) error {
+	if len(instanceOpts.Environment) == 0 {
+		log.Info("No environment variables to configure")
+		return nil
+	}
+
+	// Configure systemd system.conf.d for all services
+	// Nodeadm will use the lowest precedence directive i.e. DefaultEnvironment= to configure environment
+	// Note that EnvironmentFile= and Environment= directives will take precedence over DefaultEnvironment=.
+	// Reference: systemd.exec(5) and systemd-system.conf(5) man pages
+	// https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Environment
+	if err := writeSystemdEnvironmentConfig(log, instanceOpts.Environment); err != nil {
+		return fmt.Errorf("failed to write systemd environment config: %w", err)
+	}
+
+	for key, value := range instanceOpts.Environment {
+		if err := os.Setenv(key, value); err != nil {
+			log.Warn("Failed to set environment variable", zap.String("key", key), zap.Error(err))
+		}
+		log.Info("Set environment variable", zap.String("key", key), zap.String("value", value))
+	}
+
+	return nil
+}
+
+func writeSystemdEnvironmentConfig(log *zap.Logger, envVars map[string]string) error {
```

Comment:
I think we have a utlity method to write out a file + do the mkdir stuff? check `internal/util`